### DISCUSSION TITLE
Support for mongoose models

### DISCUSSION THREAD

**Discussion by toto1384** [colinhacks/zod#1726] [Ideas]
The only thing keeping me from moving to Zod is one project called joigoose that translates my joi models into mongoose models.

So it would be nice if one for Zod would exist. And maybe it doesn't take much time for it to be developed :))

**Comment by JacobWeisenburger** [comment]
Based on what I have read from colinhacks about his vision for zod, I highly doubt zod will natively support mongoose or any other ORM/ODM. But he is more than happy to add tools built on top of zod to the [Ecosystem](https://github.com/colinhacks/zod#ecosystem) section of the docs.

**Comment by toto1384** [comment]
After so much research, I've figured out a workaround. Turn zod to json schema and json schema to mongoose schema

The code:
```
import zodToJsonSchema from "zod-to-json-schema";
import { createMongooseSchema } from 'convert-json-schema-to-mongoose';
import { z, ZodTypeAny } from 'zod'
import mongoose from 'mongoose'

export function convertToModel(zodSchema: ZodTypeAny, collection?: string,) {
    const jsonSchema = zodToJsonSchema(zodSchema, { name: "Schema", },);

    const mongooseSchema = createMongooseSchema({}, {
        ...(jsonSchema.definitions as any).Schema,
        '$schema': 'http://json-schema.org/draft-04/schema#'
    },);

    return new mongoose.Schema(mongooseSchema, { collection });
}
```

I still hope for a proper solution
